services/src/main/java/org/keycloak/services/resources/account/AccountRestService.java

```diff
@@ -274,11 +278,22 @@ public Response sessionLogout(@QueryParam("id") String id) {
 
     @Path("/credentials")
     public AccountCredentialResource credentials() {
+        checkAccount2Enabled();
         return new AccountCredentialResource(session, event, user);
     }
 
    // TODO Federated identities
     // TODO Applications
     // TODO Logs
+    
+    private static void checkAccount2Enabled() {
+        if (!isAccount2Enabled()) {
+            throw new BadRequestException(Messages.INVALID_REQUEST);
```

Comment:
Should be 404, not found

Reply:
I'm looking at this and it turns out to be more complicated than expected.

I can throw NotFoundException, but if I do that you will actually get a 200 instead of 404.

If I don't use an exception, I can manually build and return a Response object with status of 404.  The problem there is that it won't work for endpoints that return a custom object instead of a Response object.

I see other places in the code where we throw NotFoundException and I wonder if the developer knew it was really returning 200.

There is a way to change this behavior with a RestEasy ExceptionMapper, but I don't think that's what we want to do.

WDYT?  Is it OK to just throw NotFoundException?

Reply:
Strange. NotFoundException should return a 404, that is the status code it's setting. So something weird must be happening. Is it being caught somewhere?

Reply:
Are you using javax.ws.rs.NotFoundException or some other NotFoundException?

Reply:
I've tried both the javax one and the deprecated RestEasy one.  Both returned 200.

Wondering now if someone set up an ExceptionMapper to return 200.  That would sort of make sense given how it is sometimes used in other parts of the code.

Reply:
Found something even stranger.  I can get it to return 404 if I test a different endpoint.

Even though both do the same checkAccount2Enabled(), "/sessions" endpoint returns 200 while "/credentials" endpoint returns 404.

Reply:
NotFoundException() actually does work as expected.  Apparently, it's just something weird in the way I was running the test harness.